packages/server/lib/util/settings.ts

```diff
@@ -123,16 +125,17 @@ export function isComponentTesting (options: SettingsOptions = {}) {
 }
 
 export function configFile (options: SettingsOptions = {}) {
-  // default is only used in tests.
-  // This prevents a the change from becoming bigger than it should
-  // FIXME: remove the default
-  return options.configFile === false ? false : (options.configFile || 'cypress.json')
```

Comment:
Ideally, there would not be a defult since configFile should never be null.
We could create a testing context where options always get the proper configfile resolved.

This should be done in a different PR